proto/greenfield/sp/tx.proto

```diff
@@ -148,3 +153,17 @@ message MsgUpdateParams {
 // MsgUpdateParamsResponse defines the response structure for executing a
 // MsgUpdateParams message.
 message MsgUpdateParamsResponse {}
+
+// MsgUpdateStorageProviderStatus is used to update the status of a SP by itself
+message MsgUpdateStorageProviderStatus {
+  option (cosmos.msg.v1.signer) = "sp_address";
+  // sp_address defines the operator address
+  string sp_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // status defines the desired status be update to.
+  Status status = 2;
+  // maintenancePeriod defines the time if status is set to STATUS_IN_MAINTENANCE
+  int64 maintenance_duration = 3;
```

Comment:
`maintenance_duration` seems not generic, as you can set multi kind of status, while another field is special designed for maintenance, what about just `duration`?

Reply:
changed